src/store/plugins/invites.js

```diff
@@ -0,0 +1,113 @@
+import { calculateFee, TX_TYPES } from '../../popup/utils/constants';
+import { aeToAettos, aettosToAe } from '../../popup/utils/helper';
+
+export default store =>
+  store.registerModule('invites', {
+    namespaced: true,
+    state: {
+      referrals: [],
+      client: null,
+    },
+    getters: {
+      get: state => link => state.referrals.find(r => r.link === link),
+      // eslint-disable-next-line no-unused-vars
+      keypair: state => referral => {
+        const segments = new URL(referral).pathname.split('/');
+        const secret = segments[2];
+        const { publicKey, secretKey } = Crypto.generateKeyPairFromSecret(
+          Crypto.decodeBase58Check(secret),
+        );
+
+        return {
+          publicKey: `ak_${Crypto.encodeBase58Check(Buffer.from(publicKey))}`,
+          secretKey: Buffer.from(secretKey).toString('hex'),
+        };
+      },
+      allReferrals({ referrals }, getters, { sdk }, { activeNetwork }) {
+        let { networkId } = activeNetwork;
+        if (sdk) networkId = sdk.getNetworkId();
+
+        return referrals.filter(r => r.networkId === networkId);
+      },
+    },
+    mutations: {
+      add(state, link) {
+        state.referrals.unshift(link);
+      },
+      reset(state) {
+        state.referrals = [];
+      },
+      setClient(state, client) {
+        state.client = client;
+      },
+      setBalance(state, { i, balance }) {
+        Vue.set(state.referrals, i, { ...state.referrals[i], balance });
+      },
+    },
+    actions: {
+      async claim({ rootState: { account }, dispatch, getters: { get, keypair } }, referral) {
+        const { publicKey, secretKey } = get(referral);
+        const sdkInstance = await dispatch(
+          'getClient',
+          publicKey ? { publicKey, secretKey } : keypair(referral),
+        );
+        const balance = await sdkInstance.balance(publicKey).catch(() => 0);
+        if (!balance) {
+          return false;
+        }
+        const fee = +aeToAettos(
+          calculateFee(TX_TYPES.txSign, {
+            ...sdkInstance.Ae.defaults,
+          }).min,
+        );
+        const maxSpend = balance - fee;
+        if (maxSpend <= 0) {
+          return false;
+        }
+        try {
+          const res = await sdkInstance.spend(maxSpend, account.publicKey, { payload: 'referral' });
+          return res;
+        } catch (e) {
+          dispatch('modals/open', { name: 'default', msg: e.message }, { root: true });
+          return false;
+        }
+      },
+      async getClient({ rootState: { network, current }, commit }, keypair) {
+        const { internalUrl, compilerUrl } = network[current.network];
+        const node = await Node({
+          url: internalUrl,
+          internalUrl,
+        });
+        const accounts = MemoryAccount({ keypair });
+        const sdkInstance = await Ae({
+          compilerUrl,
+          nodes: [{ name: current.network, instance: node }],
+          accounts: [accounts],
+        });
```

Comment:
No need to create sdk instance because you can set keypair to `onAccount` option (all sdk methods supports that?) and it will be used instead of the current account.
https://github.com/aeternity/aepp-sdk-js/blob/2ae757cd4c5701f3fedf3f0b5280d5862a582c19/es/account/selector.js#L48-L59

Reply:
If main sdk is used and pass `onAccount` option this line here(https://github.com/aeternity/aepp-sdk-js/blob/289598c55ff6a9d9eef8b85030e0d1f901447123/es/ae/index.js#L90) returns the address of the main account, because no `onAccount` option is used https://github.com/aeternity/aepp-sdk-js/blob/2ae757cd4c5701f3fedf3f0b5280d5862a582c19/es/account/selector.js#L64

Reply:
You are right, I have created an issue there: https://github.com/aeternity/aepp-sdk-js/issues/1046
Well, as a workaround, can we add the account by key pair, call `transferFunds` and then remove the added account?

Reply:
I changed the account but still the `address` method returns the main account `publicKey` (https://github.com/aeternity/superhero-wallet/blob/develop/src/lib/wallet.js#L90). Not sure how to handle this

Reply:
Ok, I agree to have a separate sdk instance as a workaround because of all of this.